tests/fixtures/given.go

```diff
@@ -142,30 +147,46 @@ func (g *Given) InitializeGitRepo() *Given {
 		log.Println(err)
 	}
 
-	tmpPath := filepath.Join(localPath, g.gitSync.Spec.Path)
-	dataPath := filepath.Join("testdata", g.gitSync.Spec.Path)
+	// local/repo1.git/path
+	repoNum := TrimRepoUrl(g.gitSync.Spec.RepoUrl)
+	tmpPath := filepath.Join(localPath, repoNum, g.gitSync.Spec.Path)
+	dataPath := filepath.Join("testdata", directory)
 	_ = os.Mkdir(tmpPath, 0777)
 
 	dir, err := os.ReadDir(dataPath)
 	if err != nil {
 		g.t.Fatal(err)
 	}
 
-	for _, file := range dir {
-		name := file.Name()
-		err := CopyFile(filepath.Join(dataPath, name), filepath.Join(tmpPath, name))
-		if err != nil {
-			g.t.Fatal(err)
+	for _, entry := range dir {
+		name := entry.Name()
+		if entry.IsDir() {
+			_ = os.Mkdir(filepath.Join(tmpPath, name), 0777)
+			innerDir, err := os.ReadDir(filepath.Join(dataPath, name))
+			if err != nil {
+				g.t.Fatal(err)
+			}
+			for _, dirFile := range innerDir {
+				err := CopyFile(filepath.Join(dataPath, name, dirFile.Name()), filepath.Join(tmpPath, name, dirFile.Name()))
+				if err != nil {
+					g.t.Fatal(err)
+				}
+			}
+		} else {
+			err := CopyFile(filepath.Join(dataPath, name), filepath.Join(tmpPath, name))
+			if err != nil {
+				g.t.Fatal(err)
+			}
```

Comment:
Is the idea really to recursively copy the entire directory over? Because [os.Copy() ](https://pkg.go.dev/github.com/otiai10/copy#section-readme)apparently does that.

Reply:
Looks like this is a different package but this would be exactly what we need to copy over that initial commit into the local repo folder (and replace the function I had to pull to CopyFiles as is). I had made these changes to this function for the case that we needed to test a kustomize repo which would more than likely have multiple subdirectories. The same goes for Helm so this would help a ton.

Reply:
Replaced.